pyrefly/lib/solver/subset.rs

```diff
@@ -567,6 +571,66 @@ impl<'a, Ans: LookupAnswer> Subset<'a, Ans> {
         }
     }
 
+    /// Create a ParamSpecValue type, wrapping it in Forall if it contains quantified variables
+    /// (either directly as Type::Quantified or as Type::Var backed by Variable::Quantified)
+    ///
+    /// IMPORTANT: We do NOT substitute Vars with Quantifieds here. The Vars must be preserved
+    /// so that after unification (e.g., T unified with R), we can correctly resolve all
+    /// references to the same representative Quantified in simplify_mut.
+    fn create_paramspec_value(&self, params: ParamList) -> Type {
+        // Collect all quantified variables used in the params
+        // We look for both Type::Quantified and Type::Var that are backed by Variable::Quantified
+        let mut quantifieds = Vec::new();
+
+        for param in params.items() {
+            let ty = match param {
+                Param::PosOnly(_, ty, _) => ty,
+                Param::Pos(_, ty, _) => ty,
+                Param::VarArg(_, ty) => ty,
+                Param::KwOnly(_, ty, _) => ty,
+                Param::Kwargs(_, ty) => ty,
+            };
+
+            ty.universe(&mut |t| {
+                match t {
+                    Type::Quantified(q) => {
+                        if !quantifieds.contains(&(**q)) {
+                            quantifieds.push((**q).clone());
+                        }
+                    }
+                    Type::Var(v) => {
+                        // Check if this Var is backed by a Quantified
+                        if let Some(q) = self.get_quantified_for_var(*v) {
+                            if !quantifieds.contains(&q) {
+                                quantifieds.push(q.clone());
+                            }
+                        }
+                    }
+                    _ => {}
+                }
+            });
+        }
+
+        if quantifieds.is_empty() {
+            // No quantified variables, return plain ParamSpecValue
+            Type::ParamSpecValue(params)
+        } else {
+            // Create TParams from the quantified variables
+            // Note: We keep the original Vars in the params - substitution will happen
+            // later in simplify_mut after all unifications are complete.
+            let tparams: Vec<TParam> = quantifieds
+                .into_iter()
+                .map(|q| TParam {
+                    quantified: q,
+                    variance: PreInferenceVariance::PInvariant,
```

Comment:
Is setting all the type vars to invariant safe to do? What happens if the type var is explicitly set to be covariant?